fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -2294,4 +2308,34 @@ public Pair<List<TResultBatch>, Status> executeStmtWithExecPlan(ConnectContext c
     public List<ByteBuffer> getProxyResultBuffer() {
         return proxyResultBuffer;
     }
+
+
+    // scenes can execute in FE should meet all these requirements:
+    // 1. enable_constant_execute_in_fe = true
+    // 2. is mysql text protocol
+    // 3. all values are constantOperator
+    private boolean canExecuteInFe(ConnectContext context, OptExpression optExpression) {
+        if (!context.getSessionVariable().isEnableConstantExecuteInFE()) {
+            return false;
+        }
+
+        if (context instanceof HttpConnectContext || context.getCommand() == MysqlCommand.COM_STMT_EXECUTE) {
+            return false;
+        }
+
+        if (optExpression.getOp() instanceof PhysicalValuesOperator) {
+            PhysicalValuesOperator valuesOperator = (PhysicalValuesOperator) optExpression.getOp();
+            boolean isAllConstants = true;
+            if (valuesOperator.getProjection() != null) {
```

Comment:
what if some scalar function that don't support execute in fe?

Reply:
only all constantOperator can be executed in FE. If exist other type scalarOperator, it will be executed in be.